In [1]:
import pandas as pd

# Load the dataset
file_path = 'greendestination (1).csv'
data = pd.read_csv(file_path)

# Display the first few rows of the dataset to understand its structure
data.head(), data.columns


C:\Users\vivek\anaconda3\lib\site-packages\pandas\core\computation\expressions.py:21: UserWarning: Pandas requires version '2.8.4' or newer of 'numexpr' (version '2.7.3' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED
C:\Users\vivek\anaconda3\lib\site-packages\pandas\core\arrays\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.2' currently installed).
  from pandas.core import (


(   Age Attrition     BusinessTravel  DailyRate              Department  \
 0   41       Yes      Travel_Rarely       1102                   Sales   
 1   49        No  Travel_Frequently        279  Research & Development   
 2   37       Yes      Travel_Rarely       1373  Research & Development   
 3   33        No  Travel_Frequently       1392  Research & Development   
 4   27        No      Travel_Rarely        591  Research & Development   
 
    DistanceFromHome  Education EducationField  EmployeeCount  EmployeeNumber  \
 0                 1          2  Life Sciences              1               1   
 1                 8          1  Life Sciences              1               2   
 2                 2          2          Other              1               4   
 3                 3          4  Life Sciences              1               5   
 4                 2          1        Medical              1               7   
 
    ...  RelationshipSatisfaction StandardHours  StockOption

The dataset contains various columns related to employee demographics, job characteristics, and their attrition status. Key columns of interest for this analysis include:

Age: Age of the employee.
Attrition: Whether the employee has left the company (Yes/No).
YearsAtCompany: Number of years the employee has been at the company.
MonthlyIncome: Monthly income of the employee

In [4]:
# Calculate the attrition rate
attrition_rate = data['Attrition'].value_counts(normalize=True)['Yes'] * 100

# Basic statistics for age, years at the company, and income by attrition status
attrition_analysis = data.groupby('Attrition').agg({
    'Age': ['mean', 'median'],
    'YearsAtCompany': ['mean', 'median'],
    'MonthlyIncome': ['mean', 'median']
})

attrition_rate, attrition_analysis


(16.122448979591837,
                  Age        YearsAtCompany        MonthlyIncome        
                 mean median           mean median          mean  median
 Attrition                                                              
 No         37.561233   36.0       7.369019    6.0   6832.739659  5204.0
 Yes        33.607595   32.0       5.130802    3.0   4787.092827  3202.0)

# The analysis shows:

Attrition Rate: The overall attrition rate is approximately 16.12%, meaning that about 16% of employees have left the company.

Age:

Employees who left (Yes) have an average age of about 33.6 years, with a median age of 32 years.
Employees who stayed (No) have an average age of about 37.6 years, with a median age of 36 years.
Years at Company:

Employees who left have been at the company for an average of 5.1 years, with a median of 3 years.
Employees who stayed have been at the company for an average of 7.4 years, with a median of 6 years.
Monthly Income:

Employees who left have an average monthly income of about $4,787, with a  median of $3,202.
Employees who stayed have an average monthly income of about $6,833, with  a median of $5,204.

# This suggests that younger employees, those with fewer years at the company, and those with lower incomes are more likely to leave.